ocfweb/lab_reservations/reserve.py

```diff
@@ -0,0 +1,168 @@
+        widget=forms.EmailInput(attrs={'placeholder': 'oski@berkeley.edu'}),
+    )
+
+    verify_contact_email = forms.EmailField(
+        label='Confirm contact email',
+        validators=[wrap_validator(ocflib.misc.validators.valid_email)],
+        widget=forms.EmailInput(attrs={'placeholder': 'oski@berkeley.edu'}),
+    )
+
+    student_group = forms.CharField(
+        label='Student Group',
+        widget=forms.TextInput(attrs={'placeholder': 'Open Computing Facility'}),
+        min_length=3,
+    )
+
+    reason = forms.CharField(
+        label='Reason for reservation',
+        widget=forms.Textarea(attrs={'placeholder': ''}),
+    )
+
+    date = forms.DateField(
+        label='Date of reservation (yyyy-mm-dd)',
+        widget=forms.DateInput(attrs={'placeholder': datetime.now().strftime('%Y-%m-%d')}),
+    )
+
+    starttime = forms.TimeField(
+        label='Starting time of reservation (24 hour, e.g. 22:00)',
+    )
+
+    endtime = forms.TimeField(
+        label='Ending time of reservation (24 hour, e.g. 22:00)',
+    )
+
+    disclaimer_agreement = forms.BooleanField(
+        label='I agree with the above statement.',
+        error_messages={
+            'required': 'You must agree to our policies.',
+        },
+    )
+
+    def clean_verify_(self):
+        email = self.cleaned_data.get('contact_email')
+        verify_contact_email = self.cleaned_data.get('verify_contact_email')
+
+        if email != verify_contact_email:
+            raise forms.ValidationError("Your emails don't match.")
+        return verify_contact_email
+
+
+@calnet_required
+def request_reservation(request):
+    calnet_uid = request.session['calnet_uid']
+    status = 'new_request'
+
+    # ensure we can even find them in university LDAP
+    # (alumni etc. might not be readable in LDAP but can still auth via CalNet)
+    if not user_attrs_ucb(calnet_uid):
+        return render(
+            request,
+            'lab_reservations/lab_reservations/cant-find-in-ldap.html',
+            {
+                'calnet_uid': calnet_uid,
+                'title': 'Unable to read account information',
+            },
+        )
+
+    if request.method == 'POST':
+        form = RequestForm(request.POST)
+        if form.is_valid():
+            req = NewReservationRequest(
+                real_name=form.cleaned_data['real_name'],
+                contact_email=form.cleaned_data['contact_email'],
+                student_group=form.cleaned_data['student_group'],
+                reason=form.cleaned_data['reason'],
+                date=form.cleaned_data['date'],
+                starttime=form.cleaned_data['starttime'],
+                endtime=form.cleaned_data['endtime'],
+            )
+
+            send_request_to_officers(req)
+            send_request_confirmation(req)
+
+            return redirect(reverse('request_reservation_success'))
+
+    else:
+        form = RequestForm()
+
+    return render(
```

Comment:
If you're going to have an `else` block you might as well move this statement inside it

Reply:
If we move it inside the else block we need another return statement for the case where the first if statement passes and the second fails.